com.unity.netcode.gameobjects/Tests/Runtime/NetworkManagerEventsTests.cs

```diff
@@ -0,0 +1,258 @@
+                {
+                    Assert.Fail("onClientStopped called when the client is still active");
+                }
+            };
+
+            Action<bool> onServerStopped = (bool wasAlsoClient) =>
+            {
+                callbacksInvoked++;
+                Assert.IsTrue(wasAlsoClient);
+                if (m_ServerManager.IsServer)
+                {
+                    Assert.Fail("OnServerStopped called when the server is still active");
+                }
+            };
+
+            // Start server to cause initialization process
+            Assert.True(m_ServerManager.StartHost());
+            Assert.True(m_ServerManager.IsListening);
+
+            m_ServerManager.OnServerStopped += onServerStopped;
+            m_ServerManager.OnClientStopped += onClientStopped;
+            m_ServerManager.Shutdown();
+            UnityEngine.Object.DestroyImmediate(gameObject);
+
+            yield return WaitUntilManagerShutsdown();
+
+            Assert.False(m_ServerManager.IsListening);
+            Assert.AreEqual(2, callbacksInvoked, "either OnServerStopped or OnClientStopped wasn't invoked");
+        }
+
+        [UnityTest]
+        public IEnumerator OnServerStartedCalledWhenServerStarts()
+        {
+            var gameObject = new GameObject(nameof(OnServerStartedCalledWhenServerStarts));
+            m_ServerManager = gameObject.AddComponent<NetworkManager>();
+
+            // Set dummy transport that does nothing
+            var transport = gameObject.AddComponent<DummyTransport>();
+            m_ServerManager.NetworkConfig = new NetworkConfig() { NetworkTransport = transport };
+
+            bool callbackInvoked = false;
+            Action onServerStarted = () =>
+            {
+                callbackInvoked = true;
+                if (!m_ServerManager.IsServer)
+                {
+                    Assert.Fail("OnServerStarted called when the server is not active yet");
+                }
+            };
+
+            // Start server to cause initialization process
+            m_ServerManager.OnServerStarted += onServerStarted;
+
+            Assert.True(m_ServerManager.StartServer());
+            Assert.True(m_ServerManager.IsListening);
+
+            yield return WaitUntilServerBufferingIsReady();
+
+            Assert.True(callbackInvoked, "OnServerStarted wasn't invoked");
+        }
+
+        [UnityTest]
+        public IEnumerator OnClientStartedCalledWhenClientStarts()
+        {
+            bool callbackInvoked = false;
+            Action onClientStarted = () =>
+            {
+                callbackInvoked = true;
+                if (!m_ClientManager.IsClient)
+                {
+                    Assert.Fail("onClientStarted called when the client is not active yet");
+                }
+            };
+
+            yield return InitializeServerAndAClient(onClientStarted);
+
+            Assert.True(callbackInvoked, "OnClientStarted wasn't invoked");
+        }
+        
```

Comment:
Standards Check Failure #1:
Remove the whitespaces on this line (167).

Reply:
Should be fixed now